plane2/src/types.rs

```diff
@@ -222,7 +223,7 @@ pub enum PullPolicy {
 pub struct ExecutorConfig {
     pub image: String,
     pub pull_policy: PullPolicy,
-
+    pub credentials: Option<DockerCredentials>,
```

Comment:
Let's use our own enum type for credentials rather than the bollard type, for two reasons:
- It gives us a more Rusty way of expressing required fields.
- I don't like using structs from dependencies in our public API, because it means that our API can break without us knowing if those deps change under us.

For now I think username/pass is good enough:

```rust
enum DockerRegistryAuth {
    UsernamePassword {
        username: String,
        password: String
    }
}
```